__tests__/components/QRCode/UploadQRCode.spec.ts

```diff
@@ -0,0 +1,183 @@
+            });
+
+            // Act:
+            vm.onDecode(stringJson);
+
+            // Assert:
+            expect(vm.$store.dispatch).not.toHaveBeenCalledWith('network/LOAD_TRANSACTION_FEES');
+            expect(wrapper.emitted('uploadComplete')).toEqual(undefined);
+            expect(vm.invalidType).toBe(true);
+        });
+
+        it('emits uploadComplete when valid qr types provided with fees config is defined', () => {
+            // Arrange:
+            const wrapper = getUploadQRCodeWrapper(
+                {
+                    feesConfig: networkMock.fees,
+                },
+                {
+                    validQrTypes: [QRCodeType.AddContact],
+                },
+            );
+
+            const vm = wrapper.vm as UploadQRCodeTs;
+
+            // Act:
+            vm.onDecode(stringJson);
+
+            // Assert:
+            expect(vm.$store.dispatch).not.toHaveBeenCalledWith('network/LOAD_TRANSACTION_FEES');
+            expect(wrapper.emitted('uploadComplete')).toEqual([[stringJson]]);
+            expect(vm.invalidType).toBe(false);
+        });
+    });
+
+    describe('onTabClick', () => {
+        const runBasicOnTabClickTests = (name, expectResult) => {
+            it(`set scanActive to ${expectResult} when name is ${name}`, async () => {
+                // Arrange:
+                const wrapper = getUploadQRCodeWrapper({}, {});
+
+                const vm = wrapper.vm as UploadQRCodeTs;
+
+                wrapper.setData({
+                    scanActive: !expectResult,
+                });
+
+                // Act:
+                vm.onTabClick(name);
+
+                // Assert:
+                expect(vm.scanActive).toBe(expectResult);
+            });
+        };
+
+        runBasicOnTabClickTests('scan', true);
+        runBasicOnTabClickTests('upload', false);
+    });
+
+    describe('onBeforeUpload', () => {
+        it('process image qr and pass data to qrcodeCapture', () => {
+            // Arrange:
+            const wrapper = getUploadQRCodeWrapper({}, {});
+            const vm = wrapper.vm as UploadQRCodeTs;
+
+            wrapper.setData({
+                invalidType: true,
+            });
+
+            vm.$refs.qrcodeCapture.onChangeInput = jest.fn();
+
+            const fileReaderSpy = jest.spyOn(FileReader.prototype, 'readAsDataURL').mockImplementation(() => null);
+
+            const file = {
+                name: 'invoice_qr.png',
+                size: 50000,
+                type: 'image/png',
+            };
+
+            // Act:
+            vm.onBeforeUpload(file);
+
+            // Assert:
+            expect(vm.invalidType).toBe(false);
+            expect(vm.imageFileName).toBe('invoice_qr.png');
+            expect(vm.$refs.qrcodeCapture.onChangeInput).toHaveBeenCalledWith({
+                target: {
+                    files: [file],
+                },
+            });
+            expect(fileReaderSpy).toHaveBeenCalledWith(file);
```

Comment:
check vm.image and return code

Reply:
that is an interesting part, the `fileReader.onload` is not triggered in my unit test.
I will check on this problem.